components/sidebar/sidebar_channel/sidebar_channel_link/sidebar_channel_link.tsx

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Tooltip} from 'react-bootstrap';
+import {Link} from 'react-router-dom';
+
+import {Channel} from 'mattermost-redux/types/channels';
+
+import CopyUrlContextMenu from 'components/copy_url_context_menu/copy_url_context_menu';
+import OverlayTrigger from 'components/overlay_trigger';
+
+import {mark, trackEvent} from 'actions/diagnostics_actions';
+import {localizeMessage} from 'utils/utils';
+import {isDesktopApp} from 'utils/user_agent';
+import Constants from 'utils/constants';
+import {browserHistory} from 'utils/browser_history';
+
+import ChannelMentionBadge from '../channel_mention_badge/';
+import SidebarChannelClose from '../sidebar_channel_close';
+import SidebarChannelIcon from '../sidebar_channel_icon';
+
+type Props = {
+    channel: Channel;
+    link: string;
+    label: string;
+    ariaLabelPrefix?: string;
+    closeHandler?: (callback: () => void) => void;
+    icon: JSX.Element | null;
+
+    /**
+     * Number of unread mentions in this channel
+     */
+    unreadMentions: number;
+
+    /**
+     * Number of unread messages in this channel
+     */
+    unreadMsgs: number;
+
+    /**
+     * User preference of whether the channel can be marked unread
+     */
+    showUnreadForMsgs: boolean;
+};
+
+type State = {
+    showTooltip: boolean;
+};
+
+export default class SidebarChannelLink extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            showTooltip: false,
+        };
+    }
+
+    // TODO: Gotta rethink some of this
+    // componentDidMount() {
+    //     this.enableToolTipIfNeeded();
+    // }
+
+    // componentDidUpdate(prevProps) {
+    //     if (prevProps.displayName !== this.props.displayName) {
+    //         this.enableToolTipIfNeeded();
+    //     }
+    // }
+
+    // enableToolTipIfNeeded = () => {
+    //     const element = this.displayNameRef.current;
+    //     if (element && element.offsetWidth < element.scrollWidth) {
+    //         this.setState({showTooltip: true});
+    //     } else {
+    //         this.setState({showTooltip: false});
+    //     }
+    // }
```

Comment:
No, I've decided to just re-enable the code since this was missing.
I've left a comment because I feel like we need a better way to do this, perhaps CSS can help us hide stuff.